### ISSUE TITLE
Upload of a legacy working environment?

### ISSUE THREAD

**Issue by lgbouma** [timothydmorton/VESPA#33]
Hi Tim,

This purpose of this issue is to suggest the upload of a frozen anaconda environment.yml that has as many of VESPA's dependencies as possible included, to enable dependency resolution during installation.

This could help patch over issues such as #29 #32 #13 and similar, which seem to be based around versioning issues with pandas, isochrones, and perhaps VESPA itself.

It could be created (from a working conda environment) via something similar to
`conda env export --no-builds > environment.yml`, and people could then install the environment via `conda env create -f environment.yml -n VESPA`.

I understand that certain dependencies (`MultiNest`, `pymultinest`) won't work in this approach. But it seems that this might be a helpful stopgap for users (including yours truly and a few more experienced people I've asked) who are unable to resolve these dependency conflicts!

Thank you for considering!
Luke

**Comment by lgbouma** [comment]
Hi Tim,

Good news and bad news. Good: I got an installation that seems to work. Bad: I wasn't able to figure out a single set of packages to have a seemless installation.

The installation process that seems to have worked for me was as follows.  (System info: Ubuntu 16.04, 64 bit GNU/Linux).  First, I created a dedicated environment with a specific set of dependencies that seemed like a good idea based on conversations with people who had working installations:

```
conda create --name vespa
conda install python=3.6 pandas=0.21 astropy=3.0.4 cython numba
```

I then downloaded, direct from github:

```
https://github.com/timothydmorton/isochrones/archive/1.2.2.tar.gz
https://github.com/timothydmorton/VESPA/archive/0.6.tar.gz
```

and extracted them to a local directory.

I ran `python setup.py install` for the setup.py files in each of the above isochrones and VESPA directories.

I then followed [the instructions here](https://johannesbuchner.github.io/PyMultiNest/install.html) to install multinest and PyMultiNest.

----------

This base install seemed to work reasonably well, except for one issue.  When downloading and extracting the MIST hdf5 isochrones to `~/.isochrones/mist/` (`MIST_v1.1_vvcrit0.0_UBVRIplus.h5`, `MIST_v1.1_vvcrit0.0_SDSS.h5`, `MIST_v1.1_vvcrit0.0_WISE.h5`), I encountered an error related to the column datatypes:

```
  pandas.errors.DtypeWarning: Columns (8) have mixed types. Specify dtype option on import or set low_memory=False.
```

which was being raised from line 190 of `isochrones-1.2.2/isochrones/mist/grid.py`, when trying to call the `.to_df` method to convert between the downloaded tarballs and the vespa-friendly hdf5's.

To fix this error, I tried a few things (including actually setting low_memory=False, different `pd.read_csv` engines, etc) that did not work.  The thing that worked was upgrading pandas to a "modern" version (`pandas=1.0.4`, just using `conda update pandas`), and then rerunning calcfpp to generate the h5 files.  With modern pandas, I could successfully generate the MIST hdf5 files.

However, running `calcfpp` with a modern pandas version, I would then get a different error,

```
	FPP calculation failed for ..
	Traceback (most recent call last):
		File "/home/luke/local/miniconda3/envs/vespa/lib/python3.6/site-packages/isochrones-1.2.2-py3.6.egg/isochrones/isochrone.py", line 750, in interp_value
			return interp_value(float(mass), float(age), float(feh), icol,
	TypeError: only size-1 arrays can be converted to Python scalars
```

Which seems to have been solved by reverting to pandas=0.21.

----------

So, to summarize, I did not find a single version of pandas that could be used for a clean install. However by jumping back and forth between pandas versions, I was able to get the hdf5 files extracted, and the `calcfpp` script running.

Posting this as I don't think my installation troubles have been a unique experience. Ideally I think the idea of an `environment.yml` still might be worth considering!

**Comment by timothydmorton** [comment]
@lgbouma I admire your perseverance!  At this point, if I had any free cycles to work on vespa, the priority would be to bring it up to date with isochrones >= 2.0.  I think if I (or someone else) would get around to that, that would fix all the above problems, and I think that makes a lot more sense than doing all the above gymnastics, given all the improvements in isochrones >= 2.

If you (or anyone else who still finds vespa useful) is interested in putting in a bit of work into the vespa/isochrones compatibility, I'd be happy to chat at some point to give guidance.

**Comment by lgbouma** [comment]
Understood. For the time being I'm just trying to get up to speed on the vespa basics, but I agree that this would be a very helpful upgrade for "vespa 0.7". I'll keep this in mind though if I end up using the tool more heavily. Thanks for your support!

**Comment by danhey** [comment]
Hey guys, sorry to resurrect this closed issue. I spent the better part of today trying to get `vespa` working and have managed to get it running, all the way from starfit to calcfpp. Please don't ask me how, because I'm not really sure... Anyway, I've produced the following environment, which I hope might be hopeful. 
```
name: vespa
channels:
  - conda-forge
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=4.5=1_gnu
  - ca-certificates=2021.5.30=ha878542_0
  - certifi=2021.5.30=py36h5fab9bb_0
  - cycler=0.10.0=py_2
  - freetype=2.10.4=h0708190_1
  - jbig=2.1=h7f98852_2003
  - jpeg=9d=h36c2ea0_0
  - kiwisolver=1.3.1=py36h605e78d_1
  - lcms2=2.12=hddcbb42_0
  - ld_impl_linux-64=2.35.1=h7274673_9
  - lerc=2.2.1=h9c3ff4c_0
  - libblas=3.9.0=9_openblas
  - libcblas=3.9.0=9_openblas
  - libdeflate=1.7=h7f98852_5
  - libffi=3.3=he6710b0_2
  - libgcc-ng=9.3.0=h5101ec6_17
  - libgfortran-ng=9.3.0=hff62375_19
  - libgfortran5=9.3.0=hff62375_19
  - libgomp=9.3.0=h5101ec6_17
  - liblapack=3.9.0=9_openblas
  - libopenblas=0.3.15=pthreads_h8fe5266_1
  - libpng=1.6.37=h21135ba_2
  - libstdcxx-ng=9.3.0=hd4cf53a_17
  - libtiff=4.3.0=hf544144_1
  - libwebp-base=1.2.0=h7f98852_2
  - lz4-c=1.9.3=h9c3ff4c_0
  - matplotlib-base=3.3.4=py36hd391965_0
  - mpi=1.0=mpich
  - mpich=3.4.2=h846660c_100
  - multinest=3.10=h8c654ad_8
  - ncurses=6.2=he6710b0_1
  - numpy=1.19.5=py36h2aa4a07_1
  - olefile=0.46=pyh9f0ad1d_1
  - openjpeg=2.4.0=hb52868f_1
  - openssl=1.1.1k=h7f98852_0
  - pillow=8.2.0=py36ha6010c0_1
  - pip=21.1.2=py36h06a4308_0
  - pymultinest=2.11=py36h5fab9bb_0
  - pyparsing=2.4.7=pyh9f0ad1d_0
  - python=3.6.13=h12debd9_1
  - python-dateutil=2.8.1=py_0
  - python_abi=3.6=1_cp36m
  - readline=8.1=h27cfd23_0
  - scipy=1.5.3=py36h9e8f40b_0
  - setuptools=52.0.0=py36h06a4308_0
  - six=1.16.0=pyh6c4a22f_0
  - sqlite=3.35.4=hdfb4753_0
  - tk=8.6.10=hbc83047_0
  - tornado=6.1=py36h8f6f2f9_1
  - wheel=0.36.2=pyhd3eb1b0_0
  - xz=5.2.5=h7b6447c_0
  - zlib=1.2.11=h7b6447c_3
  - zstd=1.5.0=ha95c52a_0
  - pip:
    - astropy==1.3
    - astroquery==0.4.2
    - batman-package==2.4.8
    - beautifulsoup4==4.9.3
    - cffi==1.14.5
    - chardet==4.0.0
    - corner==2.2.1
    - cryptography==3.4.7
    - cython==0.29.23
    - emcee==3.0.2
    - html5lib==1.1
    - idna==2.10
    - importlib-metadata==4.5.0
    - isochrones==1.1.1
    - jeepney==0.6.0
    - keyring==23.0.1
    - llvmlite==0.36.0
    - markupsafe==2.0.1
    - mimeparse==0.1.3
    - numba==0.53.1
    - numexpr==2.7.3
    - pandas==0.21.0
    - plotutils==0.3.2
    - pycparser==2.20
    - pytz==2021.1
    - pyvo==1.1
    - requests==2.25.1
    - secretstorage==3.3.1
    - simpledist==0.1.13
    - soupsieve==2.2.1
    - tables==3.6.1
    - typing-extensions==3.10.0.0
    - urllib3==1.26.5
    - vespa==0.5.1
    - webencodings==0.5.1
    - zipp==3.4.1
prefix: /home/daniel/anaconda3/envs/vespa
```

For reference, the conda packages were from installing pymultinest through conda-forge.

Cheers :)
Dan

**Comment by timothydmorton** [comment]
Nice---I admire your persistence!
